source/beamline/REIXS/REIXSSampleMotor.cpp

```diff
@@ -0,0 +1,377 @@
+{
+	return degrees * 0.017453292519;
+}
+
+void REIXSSampleMotor::updateMinimumAndMaximum()
+{
+	if(!isConnected()) {
+		return;
+	}
+
+	QVector3D currentGlobalPosition(horizontalTranslationControl_->value(),
+	                                normalTranslationControl_->value(),
+	                                0);
+
+	QVector3D currentPrimePosition = globalSystemToPrime(currentGlobalPosition);
+	double rotationRadians = degreesToRadians(totalRotation());
+	double sinRotation = sin(rotationRadians);
+	double cosRotation = cos(rotationRadians);
+
+	// All of the ranges have two potential limits, one constrained by the global
+	// X limit, one by the global Y limit. We calculate both, then figure out which
+	// is the most restrictive.
+	double min1 = -DBL_MAX;
+	double min2 = -DBL_MAX;
+
+	double max1 = DBL_MAX;
+	double max2 = DBL_MAX;
+
+	if(direction_ == AMMotorGroupObject::HorizontalMotion) {
+
+		// Avoid the divide by zero cases (which represents the point at which
+		// an axis is parrallel to your prime axis and the distance till you intersect
+		// it is infinite.
+		if(sinRotation == 0) {
+
+			min1 = (REIXS_SAMPLE_HORIZONTAL_MIN - currentPrimePosition.y() * sinRotation) / cosRotation;
+			max1 = (REIXS_SAMPLE_HORIZONTAL_MAX - currentPrimePosition.y() * sinRotation) / cosRotation;
+
+		} else if (cosRotation == 0) {
+
+			min2 = (currentPrimePosition.y() * cosRotation - REIXS_SAMPLE_NORMAL_MIN) / sinRotation;
+			max2 = (currentPrimePosition.y() * cosRotation - REIXS_SAMPLE_NORMAL_MIN) / sinRotation;
+
+		} else {
+			min1 = (REIXS_SAMPLE_HORIZONTAL_MIN - currentPrimePosition.y() * sinRotation) / cosRotation;
+			min2 = (currentPrimePosition.y() * cosRotation - REIXS_SAMPLE_NORMAL_MIN) / sinRotation;
+
+			max1 = (REIXS_SAMPLE_HORIZONTAL_MAX - currentPrimePosition.y() * sinRotation) / cosRotation;
+			max2 = (currentPrimePosition.y() * cosRotation - REIXS_SAMPLE_NORMAL_MIN) / sinRotation;
+
+		}
+
+	} else if (direction_ == AMMotorGroupObject::NormalMotion) {
+
+
+		if(sinRotation == 0) {
+
+			min2 = (REIXS_SAMPLE_NORMAL_MIN + currentPrimePosition.x() * sinRotation) / cosRotation;
+			max2 = (REIXS_SAMPLE_NORMAL_MAX + currentPrimePosition.x() * sinRotation) / cosRotation;
+		} else if(cosRotation == 0) {
+
+			min1 = (REIXS_SAMPLE_NORMAL_MIN - currentPrimePosition.x() * cosRotation) / sinRotation;
+			max1 = (REIXS_SAMPLE_HORIZONTAL_MAX - currentPrimePosition.x() * cosRotation) / sinRotation;
+		} else {
+
+			min1 = (REIXS_SAMPLE_NORMAL_MIN - currentPrimePosition.x() * cosRotation) / sinRotation;
+			min2 = (REIXS_SAMPLE_NORMAL_MIN + currentPrimePosition.x() * sinRotation) / cosRotation;
+
+			max1 = (REIXS_SAMPLE_HORIZONTAL_MAX - currentPrimePosition.x() * cosRotation) / sinRotation;
+			max2 = (REIXS_SAMPLE_NORMAL_MAX + currentPrimePosition.x() * sinRotation) / cosRotation;
+		}
+
+	}
+
+	// The limit is the more restrictive for each value
+	setMaximumValue(qMin(max1, max2));
+	setMinimumValue(qMax(min1, min2));
```

Comment:
I don't think AMRange would help here. I'd still have to check which is the more limiting for each of the potential min/max values. I'd end up with something like:
```C++
    ...
    AMRange newLimits(qMax(min1, min2), qMin(max1, max2));
    setMaximumValue(newLimits.minimum(), newLimits.maximum());
```

which doesn't really make things that much clearer.